pycqed/instrument_drivers/physical_instruments/DDM/DDMq.py

```diff
@@ -0,0 +1,318 @@
+                               label=('get qstate counter' +
+                                      'ch_pair {} '.format(ch_pair)),
+                               get_cmd=self._gen_ch_get_func(
+                                   self.getQstateCNT, ch_pair),
+
+                               # vals=vals.Numbers(-128,127)
+                               )
+            self.add_parameter('ch_pair{}_qstate_avg_data'.format(ch_pair),
+                               label=('get qstate average' +
+                                      'ch_pair {} '.format(ch_pair)),
+                               get_cmd=self._gen_ch_get_func(
+                                   self.getQstateAVG, ch_pair),
+
+                               # vals=vals.Numbers(-128,127)
+                               )
+        for i in range(self.device_descriptor.numChannels):
+            ch = i+1
+            self.add_parameter('ch{}_inavg_data'.format(ch),
+                               label=('get data ch {} '.format(ch)),
+                               get_cmd=self._gen_ch_get_func(
+                                   self.getInputAverage, ch),
+
+                               vals=vals.Numbers(-128, 127)
+                               )
+            srotres_cmd = 'qutech:wintrot{}:result'.format(ch)
+            self.add_parameter('ch{}_wintrot_result'.format(ch),
+                               units='#',
+                               label=('rotated integration result of' +
+                                      'channel {} '.format(ch)),
+                               get_cmd=srotres_cmd + '?'
+                               # vals=vals.Numbers(-134217728,134217727)
+                               )
+            sthl_cmd = 'qutech:qstate{}:threshold'.format(ch)
+            self.add_parameter('ch{}_qstate_threshold'.format(ch),
+                               units='#',
+                               label=('set threshold of ch {} '.format(ch)),
+                               get_cmd=sthl_cmd + '?',
+                               set_cmd=sthl_cmd + ' {}'
+                               # vals=vals.Numbers(-134217728,134217727)
+                               )
+            self.add_parameter('ch{}_weight_data'.format(ch),
+                               label=('get weigtht data ch {} '.format(ch)),
+                               get_cmd=self._gen_ch_get_func(
+                                   self.getWeightData, ch),
+                               set_cmd=self._gen_ch_set_func(
+                                   self.sendWeightData, ch),
+                               vals=vals.Arrays(-128, 127)
+                               )
+
+    def getInputAverage(self, ch):
+
+        self.write('qutech:inputavg{:d}:data? '.format(ch))
+        binBlock = self.binBlockRead()
+        print('inputavgLen in bytes {:d}'.format(len(binBlock)))
+        inputavg = np.frombuffer(binBlock, dtype=np.float32)
+        return inputavg
+
+    def getTVdata(self, ch_pair):
```

Comment:
This should also be private right ? 
Also I would replace this with get_int_avg_data